### PR TITLE
fix(coq): fix coqtop args being passed incorrectly

### LINKED ISSUE #6366: dune coq top ignores flags specified in coq.theory
## Expected Behavior

```dune coq top``` should pass the flags specified in the ```(coq.theory (flags ...))``` to coqtop / the toplevel.

## Actual Behavior

Flags specified in the dune file are ignored by ```dune coq top``` and not passed on to ```coqtop```.

## Reproduction

1. Create a dune file with contents
```
(coq.theory
  (name minimal)
  (flags -w -notation-overridden))
```
2. Create a file ```Test.v``` with contents ```Notation "a + b" := (Nat.mul a b) : nat_scope.```

3. The arguments in the dune file is then not passed along to ```coqtop```:
```
$ dune coq top --display=quiet --toplevel=echo Test.v
-topfile /tmp/minimal/_build/default/Test.v -R /tmp/minimal/_build/default minimal
```
for example, we get a warning here: 
```
$ cat Test.v | dune coq top --display=quiet Test.v
Welcome to Coq 8.16.0

Coq < Toplevel input, characters 0-46:
> Notation "a + b" := (Nat.mul a b) : nat_scope.
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Warning: Notation "_ + _" was already used in scope nat_scope.
[notation-overridden,parsing]
```
## Specifications

- Version of `dune` (output of `dune --version`): ```3.5.0```
- Version of `ocaml` (output of `ocamlc --version`): ```4.14.0```
- Operating system (distribution and version):

## Additional information 

- Link to gist with verbose output (run `dune` with the `--verbose` flag):

### PR DESCRIPTION
**Pull request by Alizter**
fixes #6366 

More refactoring of coqtop will have to come at a later date, this at least makes the flags match up correctly.

### CODE REVIEW

**Comment by ejgallego** [comment]
> I've undone a previous refactoring PR splitting the functions up since it was basically completely pointless.

Please do that in a separate PR or commit as it makes reviewing the code changes very hard.

**Comment by Alizter** [comment]
@ejgallego The diff should be cleaner now.

**Comment by Alizter** [comment]
@ejgallego ping

**Review by ejgallego — APPROVED** [review]
Looks good to me, thanks for this fix; but we need to refactor the way flags are setup, now that we have multiple users, the current API it is indeed pretty error prone.

**Comment by ejgallego** [comment]
This deserves an entry in `CHANGES.md` tho.

File: CHANGES.md [hunk 1]
```diff
@@ -52,6 +52,8 @@ Unreleased
 - Allow `:standard` in the `(modules)` field of the `coq.pp` stanza (#6229,
   fixes #2414, @Alizter)
 
+- Fix passing of native flags to dune coq top (#6369, fixes #6366, @Alizter)
```
Comment (ejgallego) [inline CHANGES.md:55]: Not only native, but user flags too (that was actually the bug IIRC)

**Pull request merged**